core/src/main/java/org/jdbi/v3/Query.java

```diff
@@ -76,7 +76,7 @@
 
     @Override
     public <ContainerType> ContainerType collectInto(Class<ContainerType> containerType) {
-        return collect(getCollectorFactoryRegistry().createCollectorFor(containerType));
+        return collect(getCollectorFactoryRegistry().createCollectorFor(getHandle().getTypeResolver().resolve(containerType)));
```

Comment:
It does. Which is why I'd rather pivot on the fluent interfaces and only provide APIs which can make type safety guarantees.

I'd rather that collector factories were only used behind the scenes with SQL objects.